doc/configuration.rst

```diff
@@ -270,40 +270,64 @@ The actual content (RST content with images, formulas, etc) of the item is curre
 Callback per item (advanced)
 ----------------------------
 
+Callback to modify item
+=======================
+
 The plugin allows parsing and modifying documentation objects *behind the scenes* using a callback. The callback
 has this prototype:
 
 .. code-block:: python
 
     def traceability_callback_per_item(name, collection):
-        ''' Custom callback on items
+        """Callback function called when an item-directive is being processed.
+
+        Note: attributes, relationships and content (body) of the item can be modified. Sphinx processes each directive
+        in turn, so attributes and relationships added or modified by other directives may not have been processed yet.
 
         Args:
             name (str): Name (id) of the item currently being parsed
             collection (TraceableCollection): Collection of all items that have been parsed so far
-        '''
+        """
         pass
 
-The callback is executed while parsing the documentation item from your rst-file. Note that not all items are
-available at the time this callback executes, the *collection* parameter is a growing collection of documentation
-objects.
+.. note::
+
+    The callback is executed while parsing the documentation item from your RST file. Note that not all items are
+    available at the time this callback executes, the *collection* parameter is a growing collection of documentation
+    objects.
 
-.. _traceability_no_callback:
+Callback to inspect item
+========================
 
-Example of no callback per item
-===============================
+To overcome the limitation of ``traceability_callback_per_item`` (see note above), a secondary callback function can be
+defined. This function will be called when *rendering* each ``item``-directive. At that moment, all other directive
+types, e.g. ``attribute-link`` and ``item-link``, will have been processed. You can use this callback function to detect
+and warn about any gaps in your documentation but you shouldn't use it to make any modifications.
+The callback has this prototype:
 
 .. code-block:: python
 
-    traceability_callback_per_item = None
+    def traceability_inspect_item(name, collection):
+        """Callback function called when an item-directive is being rendered.
+
+        Note: the item should not be modified, only inspected. At this stage of the documentation build, all directives,
+        e.g. attribute-link and item-link, have been processed and any gaps in your documentation can be exposed by
+        reporting a warning.
+
+        Args:
+            name (str): Name (id) of the item currently being parsed
+            collection (TraceableCollection): Collection of all items that have been parsed so far
+        """
+        pass
```

Comment:
```suggestion
        pass
        
.. warning::

    Collection should not be modified, only inspected. Modifying collection in this step can corrupt it without triggering any warnings.
```

Reply:
Added warning in 42676d558a07db3a81ea9be33b9ace95621c6ce6